File: library/src/main/java/com/github/jura73/materialspinner/ViewHelper.java
package com.github.jura73.materialspinner;

import android.app.Dialog;
import android.content.Context;
import android.support.v7.widget.SearchView;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Filter;

class ViewHelper {
    static void setupToolbar(final Dialog dialog, final Filter filter) {
        Toolbar toolbar = dialog.findViewById(R.id.toolbar);
        toolbar.inflateMenu(R.menu.menu_main);
        toolbar.setTitle(R.string.action_search);
        toolbar.setNavigationIcon(dialog.getContext().getResources().getDrawable(R.drawable.ic_arrow_back_white_24dp));
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialog.onBackPressed();
            }
        });

        SearchView searchView = (SearchView) toolbar.getMenu().findItem(R.id.action_search)
                .getActionView();
        searchView.setMaxWidth(Integer.MAX_VALUE);
        searchView.setQueryHint(dialog.getContext().getString(R.string.action_search));
        // expands SearchView
        searchView.setIconified(false);
        searchView.clearFocus();
        // not closes the SearchView
        searchView.setOnCloseListener(new SearchView.OnCloseListener() {
            @Override
            public boolean onClose() {
                return true;
            }
        });
        // listening to search query text change
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                filter.filter(query);
                return false;
            }

            @Override
            public boolean onQueryTextChange(String query) {
                filter.filter(query);
                return false;
            }
        });
    }

    static void hideSoftInput(final Dialog dialog){
        View view = dialog.getCurrentFocus();
        if (view != null) {
            InputMethodManager imm = (InputMethodManager)dialog.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
            imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
        }
    }
}

File: library/src/main/java/com/github/jura73/materialspinner/ViewHelper.kt
package com.github.jura73.materialspinner

import android.app.Dialog
import android.content.Context
import android.support.v7.widget.SearchView
import android.support.v7.widget.Toolbar
import android.view.inputmethod.InputMethodManager
import android.widget.Filter

internal object ViewHelper {
    fun setupToolbar(dialog: Dialog, filter: Filter) {
        val toolbar = dialog.findViewById<Toolbar>(R.id.toolbar)
        toolbar.inflateMenu(R.menu.menu_main)
        toolbar.setNavigationOnClickListener { dialog.onBackPressed() }

        val searchView = toolbar.menu.findItem(R.id.action_search).actionView as SearchView
        searchView.maxWidth = Integer.MAX_VALUE
        // expands SearchView
        searchView.isIconified = false
        searchView.clearFocus()
        searchView.queryHint = dialog.context.getString(R.string.action_search)
        // not closes the SearchView
        searchView.setOnCloseListener { true }
        // listening to search query text change
        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                filter.filter(query)
                return false
            }

            override fun onQueryTextChange(query: String): Boolean {
                filter.filter(query)
                return false
            }
        })
    }

    fun hideSoftInput(dialog: Dialog) {
        val view = dialog.currentFocus
        if (view != null) {
            val imm = dialog.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }
    }
}